src/HttpClient/Plugin/Authentication/HybridOauth2.php

```diff
@@ -0,0 +1,175 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Apigee\Edge\HttpClient\Plugin\Authentication;
+
+use Apigee\Edge\Client;
+use Apigee\Edge\ClientInterface;
+use Apigee\Edge\Exception\HybridOauth2AuthenticationException;
+use Firebase\JWT\JWT;
+use Http\Client\Exception;
+use Http\Message\Authentication;
+use Http\Message\Authentication\Bearer;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * HybridOauth2 authentication plugin for authenticating to Hybrid Cloud API.
+ *
+ * @see https://developers.google.com/identity/protocols/OAuth2ServiceAccount
+ */
+class HybridOauth2 implements Authentication
+{
+    /**
+     * Authorization server for Apigee Hybrid authentication.
+     *
+     * @var string
+     */
+    public const DEFAULT_AUTHORIZATION_SERVER = 'https://oauth2.googleapis.com/token';
+
+    /**
+     * A space-delimited list of the permissions that the application requests.
+     *
+     * @var string
+     */
+    public const TOKEN_SCOPES = 'https://www.googleapis.com/auth/cloud-platform';
+
+    /**
+     * Grant type used in an access token request.
+     *
+     * @see https://developers.google.com/identity/protocols/OAuth2ServiceAccount
+     *
+     * @var string
+     */
+    public const GRANT_TYPE = 'urn:ietf:params:oauth:grant-type:jwt-bearer';
+
+    /**
+     * The service account email.
+     *
+     * @var string
+     */
+    protected $email;
+
+    /**
+     * The service account private key.
+     *
+     * @var string
+     */
+    protected $private_key;
+
+    /**
+     * @var \Apigee\Edge\HttpClient\Plugin\Authentication\OauthTokenStorageInterface
+     */
+    protected $tokenStorage;
+
+    /**
+     * @var string
+     */
+    protected $auth_server;
+
+    /**
+     * Hybrid Oauth2 constructor.
+     *
+     * @param string $email
+     *   The service account email.
+     * @param string $private_key
+     *   The service account private key.
+     * @param \Apigee\Edge\HttpClient\Plugin\Authentication\OauthTokenStorageInterface $token_storage
+     *   Storage where access token gets saved.
+     * @param string|null $auth_server
+     *   Authentication server.
+     */
+    public function __construct(string $email, string $private_key, OauthTokenStorageInterface $token_storage, ?string $auth_server = null)
+    {
+        $this->email = $email;
+        $this->private_key = $private_key;
+        $this->tokenStorage = $token_storage;
+        $this->auth_server = $auth_server ?: self::DEFAULT_AUTHORIZATION_SERVER;
```

Comment:
camelCase, yes it should be camelCased in the Oauth plugin as well :-(

Reply:
Fixed.